src/plugins/roam/roam.js

```diff
@@ -236,19 +247,35 @@
         }
     }
 
-    // Init `roams.keys` with `""`, which means no key's pressed down.
-    roams.keys = "";
-    Array.from( roamKeys ).forEach( ( key ) => {
-        roams[ key ] = {
-            start: undefined
-        };
-    } );
+    document.addEventListener( "impress:init", function() {
+        // Update `api` and `gc` to current ones.
+        api = window.impress();
+        gc = api.lib.gc;
 
-    // Request to be called before next frame renders.
-    // And every single frame after is going to be called in the `roamAnimationFrame()` function.
-    window.requestAnimationFrame( roamAnimationFrame );
+        // Stop requesting animation frame when `impress().tear()` is called.
+        api.tear = new Proxy( api.tear, {
+            apply ( target, thisArg, argumentsList ) {
+                impressActive = false;
+                target.apply( thisArg, argumentsList );
+            }
+        } );
```

Comment:
Ah, sorry. I should have pointed to this in my last review, but I had forgotten how exactly to do this. Please use

    gc.pushCallback(my_function);

https://github.com/impress/impress.js/blob/f68e5fff298469cf9836fa6121f665d8ab323e2a/src/lib/gc.js#L60

... to register a function that is called as part of `impress().tear()`.

Example:

https://github.com/impress/impress.js/blob/d0c427f172453b6306fb88d638df7d1594c476c8/src/plugins/progress/progress.js#L22